### PR TITLE
bug/VOGRE-13

### PR DESCRIPTION
**Pull request by Girik1105**
# Guidelines for Pull Requests

If you haven't yet read our code review guidelines, please do so, You can find them [here](https://diging.atlassian.net/wiki/spaces/DIGING/pages/2256076801/Code+Review+Guidelines).

Please confirm the following by adding an x for each item (turn `[ ]` into `[x]`).

- [x] I have removed all code style changes that are not necessary (e.g. changing blanks across the whole file that don’t need to be changed, adding empty lines in parts other than your own code)
- [x] I am not making any changes to files that don’t have any effect (e.g. imports added that don’t need to be added)
- [x] I do not have any sysout statements in my code or commented out code that isn’t needed anymore
- [x] I am not reformatting any files in the wrong format or without cause. 
- [x] I am not changing file encoding or line endings to something else than UTF-8, LF
- [x] My pull request does not show an insane amount of files being changed although my ticket only requires a few files being changed
- [x] I have added Javadoc/documentation where appropriate
- [x] I have added test cases where appropriate
- [x] I have explained any part of my code/implementation decisions that is not be self-explanatory

## Please provide a brief description of your ticket 

Importing texts: text list seems to only show the first 50 texts

Description:
not all of them

https://diging.atlassian.net/browse/VOGRE-13?atlOrigin=eyJpIjoiYWIyZjNhN2JkZWZmNGM1NTkzZjQ3MjhiNTBlMzdmY2IiLCJwIjoiaiJ9

## Are there any other pull requests that this one depends on?
No
  
## Anything else the reviewer needs to know?

... describe here ...

### CODE REVIEW

File: repository/managers.py [hunk 1]
```diff
@@ -40,18 +40,70 @@ def collections(self, groupId):
...
+            collections_response.raise_for_status()
+
+        # Parse the response to find the specific collection and get the number of items in the collection
+        collections_data = collections_response.json().get('collections', [])
+        collection_num_items = 0
+        group_info = collections_response.json().get('group', {})
+        for collection in collections_data:
+            if collection.get('key') == collectionId:
+                collection_num_items = collection.get('numberOfItems', 0)
+                break
```
Comment (jdamerow) [inline repository/managers.py]: alright, we need to change that. there needs to be a citesphere endpoint to get the information about a specific collection. let me see if I merge now or wait till that is there.

File: repository/managers.py [hunk 2]
```diff
@@ -40,18 +40,70 @@ def collections(self, groupId):
...
+
+        # Get total pages which will be required to get all items as it only returns 50 in one request
+        total_pages = (collection_num_items // 50) + (1 if collection_num_items % 50 else 0)
```
Comment (jdamerow) [inline repository/managers.py]: 50 needs to be a value from settings. it might change.

File: repository/managers.py [hunk 3]
```diff
@@ -40,18 +40,70 @@ def collections(self, groupId):
...
+        response = requests.get(f"{base_url}?page=1", headers=headers)
+        if response.status_code != 200:
             response.raise_for_status()
 
+        # Add the items from the first page to the final result
+        first_page_data = response.json()
+        final_result["items"].extend(first_page_data.get('items', []))
+
+        # Fetch subsequent pages if there are more than one
+        for page in range(2, total_pages + 1):
```
Comment (jdamerow) [inline repository/managers.py]: only one page at a time should be retrieved from citesphere, not all.

**Review by jdamerow — CHANGES_REQUESTED** [review]

File: repository/managers.py [hunk 4]
```diff
@@ -40,17 +40,47 @@ def collections(self, groupId):
...
+        """
         headers = auth.citesphere_auth(self.user, self.repository)
-        url = f"{self.repository.endpoint}/api/v1/groups/{groupId}/collections/{collectionId}/items/"
-        response = requests.get(url, headers=headers)
         
-        if response.status_code == 200:
-            return response.json()
-        else:
-            response.raise_for_status()
+        base_url = f"{self.repository.endpoint}/api/v1/groups/{groupId}/collections/{collectionId}/items/"
```
Comment (jdamerow) [inline repository/managers.py]: that's not the base url but the url to get the items from, isn't it?

File: repository/managers.py [hunk 5]
```diff
@@ -40,17 +40,47 @@ def collections(self, groupId):
...
+        collections_url = f"{self.repository.endpoint}/api/v1/groups/{groupId}/collections/"
+
+        # Fetch collection details to get total items
+        collections_response = requests.get(collections_url, headers=headers)
+        collections_response.raise_for_status()
+
+        # Extract group and total items for the collection
+        collections_data = collections_response.json().get('collections', [])
+        group_info = collections_response.json().get('group', {})
+        total_items = next((c.get('numberOfItems', 0) for c in collections_data if c.get('key') == collectionId), 0)
```
Comment (jdamerow) [inline repository/managers.py:73]: let's add a todo here that this needs to be changed once there is a collection info endpoint

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Comment by jdamerow** [comment]
not sure what's going wrong, but I don't see any texts, just collections when I go to a group (texts do show up in a collection)

**Comment by Girik1105** [comment]
> not sure what's going wrong, but I don't see any texts, just collections when I go to a group (texts do show up in a collection)

The texts in a group is another ticket [VOGRE-10](https://diging.atlassian.net/browse/VOGRE-10?atlOrigin=eyJpIjoiZmVjYmM1MjQ1NjQxNGJmNWI1YWIxMmViZDRhYmJhMzciLCJwIjoiaiJ9) 

[VOGRE-10]: https://diging.atlassian.net/browse/VOGRE-10?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by jdamerow** [comment]
ah,ok.

**Pull request merged**